src/main/java/core/basesyntax/dao/animal/AnimalDaoImpl.java

```diff
@@ -12,11 +14,33 @@ public AnimalDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Animal save(Animal animal) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(animal);
+            transaction.commit();
+            return animal;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Unable to add animal: " + animal, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Animal> findByNameFirstLetter(Character character) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Animal"
+                    + "where name like :name", Animal.class)
```

Comment:
I received a comment about using LIKE with the matching pattern only
Not a rule, more of a suggestion
So let's replace it with "="

Reply:
We use% to search for a word, we can only do this with the LIKE operator

Reply:
Sorry, I thought it was a findBuColor method 🤦‍♀️